h2o-automl/src/main/java/ai/h2o/automl/modeling/XGBoostSteps.java

```diff
@@ -0,0 +1,188 @@
+                    xgBoostParameters._sample_rate = 0.8;
+                    xgBoostParameters._col_sample_rate = 0.8;
+                    xgBoostParameters._col_sample_rate_per_tree = 0.8;
+
+                    if (_emulateLightGBM) {
+                        xgBoostParameters._max_leaves = 1 << xgBoostParameters._max_depth;
+                        xgBoostParameters._max_depth = xgBoostParameters._max_depth * 2;
+//                        xgBoostParameters._min_data_in_leaf = (float) xgBoostParameters._min_rows;
+                        xgBoostParameters._min_sum_hessian_in_leaf = (float) xgBoostParameters._min_rows;
+                    }
+
+                    return trainModel(xgBoostParameters);
+                }
+            },
+    };
+
+    private ModelingStep[] grids = new XGBoostGridStep[] {
+            new XGBoostGridStep("grid_1", 5* DEFAULT_GRID_TRAINING_WEIGHT, aml(), false) {
+                @Override
+                protected Job<Grid> startJob() {
+                    XGBoostParameters xgBoostParameters = prepareModelParameters();
+                    Map<String, Object[]> searchParams = new HashMap<>();
+//                    searchParams.put("_ntrees", new Integer[]{100, 1000, 10000}); // = _n_estimators
+
+                    if (_emulateLightGBM) {
+                        searchParams.put("_max_leaves", new Integer[]{1<<5, 1<<10, 1<<15, 1<<20});
+                        searchParams.put("_max_depth", new Integer[]{10, 20, 50});
+                        searchParams.put("_min_sum_hessian_in_leaf", new Double[]{0.01, 0.1, 1.0, 3.0, 5.0, 10.0, 15.0, 20.0});
+                    } else {
+                        searchParams.put("_max_depth", new Integer[]{5, 10, 15, 20});
+                        searchParams.put("_min_rows", new Double[]{0.01, 0.1, 1.0, 3.0, 5.0, 10.0, 15.0, 20.0});  // = _min_child_weight
+                    }
+
+                    searchParams.put("_sample_rate", new Double[]{0.6, 0.8, 1.0}); // = _subsample
+                    searchParams.put("_col_sample_rate" , new Double[]{ 0.6, 0.8, 1.0}); // = _colsample_bylevel"
+                    searchParams.put("_col_sample_rate_per_tree", new Double[]{ 0.7, 0.8, 0.9, 1.0}); // = _colsample_bytree: start higher to always use at least about 40% of columns
+//                    searchParams.put("_learn_rate", new Double[]{0.01, 0.05, 0.08, 0.1, 0.15, 0.2, 0.3, 0.5, 0.8, 1.0}); // = _eta
+//                    searchParams.put("_min_split_improvement", new Float[]{0.01f, 0.05f, 0.1f, 0.5f, 1f, 5f, 10f, 50f}); // = _gamma
+//                    searchParams.put("_tree_method", new XGBoostParameters.TreeMethod[]{XGBoostParameters.TreeMethod.auto});
+                    searchParams.put("_booster", new XGBoostParameters.Booster[]{ //gblinear crashes currently
+                            XGBoostParameters.Booster.gbtree, //default, let's use it more often
+                            XGBoostParameters.Booster.gbtree,
```

Comment:
does GridSearch skip duplicate parameters? eg. ntrees=5, booster=gbtree and second ntrees=5, booster=gbtree'

Reply:
damn, you're right, random search keeps track of indices combinations, not of parameters combinations...
I guess the harm is not big today, as the hyperspace is large, but I need to reconsider this: I wish there was a way to give weights to some param, but as you suggested above, maybe the best here is to have a separate grid, though I'm not sure it's worth having a full grid with similar search space for each booster.
Will definitely revisit this.

Reply:
https://0xdata.atlassian.net/browse/PUBDEV-7289